qiita_db/test/test_metadata_template.py

```diff
@@ -1214,6 +1214,77 @@ def test_get_filepath(self):
         st = SampleTemplate.create(self.metadata, self.new_study)
         self.assertEqual(st.get_filepaths()[0][0], exp_id)
 
+    def test_extend(self):
+        # add new column and delete one that exists
+        self.metadata['NEWCOL'] = pd.Series(['val1', 'val2', 'val3'],
+                                            index=self.metadata.index)
+        self.tester.extend(self.metadata)
+
+        # test samples were appended successfully
+        sql = ("SELECT sample_id FROM qiita.required_sample_info WHERE "
+               "study_id = 1")
+        obs = self.conn_handler.execute_fetchall(sql)
+        exp = [['1.SKB8.640193'], ['1.SKD8.640184'], ['1.SKB7.640196'],
+               ['1.SKM9.640192'], ['1.SKM4.640180'], ['1.SKM5.640177'],
+               ['1.SKB5.640181'], ['1.SKD6.640190'], ['1.SKB2.640194'],
+               ['1.SKD2.640178'], ['1.SKM7.640188'], ['1.SKB1.640202'],
+               ['1.SKD1.640179'], ['1.SKD3.640198'], ['1.SKM8.640201'],
+               ['1.SKM2.640199'], ['1.SKB9.640200'], ['1.SKD5.640186'],
+               ['1.SKM3.640197'], ['1.SKD9.640182'], ['1.SKB4.640189'],
+               ['1.SKD7.640191'], ['1.SKM6.640187'], ['1.SKD4.640185'],
+               ['1.SKB3.640195'], ['1.SKB6.640176'], ['1.SKM1.640183'],
+               ['1.Sample1'], ['1.Sample2'], ['1.Sample3']]
+        self.assertEqual(obs, exp)
+
+        sql = "SELECT sample_id FROM qiita.sample_1"
+        obs = self.conn_handler.execute_fetchall(sql)
+        exp = [['1.SKM7.640188'], ['1.SKD9.640182'], ['1.SKM8.640201'],
+               ['1.SKB8.640193'], ['1.SKD2.640178'], ['1.SKM3.640197'],
+               ['1.SKM4.640180'], ['1.SKB9.640200'], ['1.SKB4.640189'],
+               ['1.SKB5.640181'], ['1.SKB6.640176'], ['1.SKM2.640199'],
+               ['1.SKM5.640177'], ['1.SKB1.640202'], ['1.SKD8.640184'],
+               ['1.SKD4.640185'], ['1.SKB3.640195'], ['1.SKM1.640183'],
+               ['1.SKB7.640196'], ['1.SKD3.640198'], ['1.SKD7.640191'],
+               ['1.SKD6.640190'], ['1.SKB2.640194'], ['1.SKM9.640192'],
+               ['1.SKM6.640187'], ['1.SKD5.640186'], ['1.SKD1.640179'],
+               ['1.Sample1'], ['1.Sample2'], ['1.Sample3']]
+        self.assertEqual(obs, exp)
+
+        # test new columns were added to *_cols table and dynamic table
```

Comment:
oh thanks, since you didn't modify `self.metadata` to add new columns I was confused.